### ISSUE TITLE
cabal2nix in nixpkgs 18.03 is broken for packages with Unicode in .cabal file

### ISSUE THREAD

**Issue by Taneb** [NixOS/nixpkgs#39806]
## Issue description

`nixpkgs` 18.03 includes `cabal2nix` version 2.9. This version has a bug which prevents using it with any package with non-ASCII characters in its .cabal file from a local directory and from a git URI. This includes `clash-prelude` and, alarmingly, `cabal2nix`.

`master` of `nixpkgs` includes `cabal2nix` version 2.9.2 which fixes this issue. Including this version of `cabal2nix` in 18.03 would solve this issue.

### Steps to reproduce

To demonstrate the issue with the `cabal2nix` in 18.03, I did the following steps:

```sh
cabal unpack cabal2nix-2.9
nix-build cabal2nix
result/bin.cabal2nix cabal2nix-2.9/
```

This prints:
```
*** parsing cabal file: user error (cannot parse "./cabal2nix-2.9/cabal2nix.cabal")
cabal2nix: nix-prefetch-zip: createProcess: runInteractiveProcess: exec: does not exist (No such file or directory)
```

## Technical details

I am using using Nix version 1.11.9 and `nixpkgs` revision 4647755b9fd44f37ce6361597c7ac4694d14a742

**Comment by peti** [comment]
Fixed in https://github.com/NixOS/nixpkgs/issues/39806. I updated only `cabal2nix`, not all of Hackage, which would feel a bit too risky for a stable branch.

**Comment by peti** [comment]
Fixed in aa0c2d59fc811850355d898fda8eb27fd3031689. I updated only cabal2nix, not all of Hackage, which would feel a bit too risky for a stable branch.
